File: projects/engineering.purdue.edu/kak/programmingwithobjects/pwocode/chap10/ExceptionUsage2.cc
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Book Author:   Avinash Kak
//
// Chapter:     Chapter 10  Handling Exceptions
//
// Section:     Section 10.3  Some Usage Patterns For Exceptions Handling In C++
//



//ExceptionUsage2.cc

#include <iostream>
using namespace std;

class MyException {};

void f() { throw MyException(); }                                 //(A)

int main()
{
    try {
        f();                                                      //(B)
    } catch( MyException ) {                                      //(C)
        cout << "caught MyException";            
    }

    return 0;
}

File: projects/engineering.purdue.edu/kak/programmingwithobjects/pwocode/chap10/ExceptionUsage2.java
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Book Author:   Avinash Kak
//
// Chapter:     Chapter 10  Handling Exceptions
//
// Section:     Section 10.6  Some Usage Patterns For Exception Handling In Java
//



//ExceptionUsage2.java

class MyException extends Exception {}

class Test {
    static void f( ) throws MyException {}

    public static void main( String[] args ) {
        try {
            f();
        } catch( MyException e ) {}
    }
}